### ISSUE TITLE
Fix engine crash under SSj due to mismatched JsRT script IDs

### ISSUE THREAD

**Issue by fatcerberus** [spheredev/neosphere#222]
I thought Chakra assigned ScriptIds sequentially, but there have been a few cases in in-house testing where miniSphere has sent the wrong filename to SSj for a module.  Figure out what's causing that, as the script ID handling seems correct at a glance.

**Comment by fatcerberus** [comment]
Marked this as a bug since it causes SSj to sometimes show the wrong filename and source listing for a script, compromising the user experience.

**Comment by fatcerberus** [comment]
From what I've been able to gather, script IDs are assigned by the debugger in the same order the corresponding scripts were compiled.  Script IDs are indeed sequential; however, *every single compiler invocation* gets its own script ID, including those scripts compiled through the JsRT API using `JS_SOURCE_CONTEXT_NONE`.

In implementing the debugger, I was able to work around this pretty well by incrementing an internal counter any time JavaScript code is passed to a JsRT API and associating the current with a filename when applicable.  This works--until a script calls either `eval()` or `new Function()`, which invokes the compiler and knocks miniSphere's internal script counter out of sync.

A more robust solution will be needed, preferably before miniSphere 5.0.0 is released.
